nomad/leader.go

```diff
@@ -519,6 +522,52 @@ func (s *Server) periodicUnblockFailedEvals(stopCh chan struct{}) {
 	}
 }
 
+// publishJobSummaryMetrics publishes the job summaries as metrics
+func (s *Server) publishJobSummaryMetrics(stopCh chan struct{}) {
+	// Using a timer instead of a ticker so that we can publish after the
+	// current batch of metrics have been published
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-stopCh:
+			return
+		case <-timer.C:
+			state, err := s.State().Snapshot()
+			if err != nil {
+				timer.Reset(s.config.StatsCollectionInterval)
+				s.logger.Printf("[ERR] nomad: failed to get state: %v", err)
+				continue
+			}
+			ws := memdb.NewWatchSet()
+			iter, err := state.JobSummaries(ws)
+			if err != nil {
+				timer.Reset(s.config.StatsCollectionInterval)
+				s.logger.Printf("[ERR] nomad: failed to get job summaries: %v", err)
+				continue
+			}
+
+			for {
+				raw := iter.Next()
+				if raw == nil {
+					break
+				}
+				summary := raw.(*structs.JobSummary)
+				for name, tgSummary := range summary.Summary {
+					metrics.SetGauge([]string{"nomad", "job_summary", summary.JobID, name, "queued"}, float32(tgSummary.Queued))
```

Comment:
As of Nomad 0.7, metrics are published in a tagged format. See: 
- https://www.nomadproject.io/api/metrics.html
- https://www.nomadproject.io/docs/agent/configuration/telemetry.html#backwards_compatible_metrics

Reply:
Yeah I saw that, but I wasn't sure what the tags are going to be in this case.

Reply:
Any suggestions?

Reply:
I think the job should be a tag.